src/axom/spin/ImplicitGrid.hpp

```diff
@@ -204,8 +204,20 @@ class ImplicitGrid
     for(int i = 0; i < NDIMS; ++i)
     {
       m_bins[i] = BinSet(m_gridRes[i]);
-      m_binData[i] =
-        BinBitMap(&m_bins[i], BitsetType(numElts, allocatorID), 1, allocatorID);
```

Comment:
The issue is in this line - when each bin in the bin map is filled with copies of `slam::BitSet`, this copy constructor is called for each `BitSet`'s underlying word array:
https://github.com/LLNL/axom/blob/5f7869f21c4a2ee0ab0756e6f241690db6bcde86/src/axom/core/Array.hpp#L182-L188
Importantly, this constructor doesn't propagate allocator IDs, so even though we're passing a `BitSet` allocated in device memory, the resulting bitset in each bin is allocated in the default "dynamic" space - host memory.

Reply:
Thanks for the pointer @kanye-quest 

Do we have a constructor that does propagate the allocator ID? and should we be using that here?